s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```diff
@@ -63,10 +102,17 @@ Iterator<S3ObjectSummary> fetchObjectSummaries(final AmazonS3 s3Client) {
                         .stream()
                         .filter(objectSummary -> objectSummary.getSize() > 0)
                         .filter(objectSummary -> assignObjectToTask(objectSummary.getKey()))
-                        .filter(objectSummary -> !failedObjectKeys.contains(objectSummary.getKey())));
+                        .filter(objectSummary -> !failedObjectKeys.contains(objectSummary.getKey())))
+                .map(S3ObjectSummary::getKey);
```

Comment:
I am looking at this now, it looks like we, basically load all of the ObjectListKeys into the iterator at the start in one go.
We can make an update to lazy load this, so that the maximum number of keys are basically returned and processed (configurable but default and maximum is 1,000).
We can then see when the results are all returned, and call getListofObjectKeys(startAfter) and get the next batch of results?


WDYT?